R/functions_degs.R

```diff
@@ -53,6 +53,7 @@ DegsUpDisplayTop = function(degs, n=5, caption=NULL) {
   top = degs %>% 
     dplyr::group_by(cluster) %>% 
     dplyr::top_n(n=n, wt=p_val_adj_score) %>% 
+    dplyr::top_n(n=n, wt=pct.1-pct.2) %>%
```

Comment:
My intention was to deal with cases, when lets say 10 genes have `p_adj = 0`, but we want to pick the top 5 only. Since we look at `p_val_adj_score`, the top 10 automatically have a positive FC, so no need for `abs`. We could also go for FC. I thought this percentage difference is nice because it is not about the expression values, but about the cells that actually express the gene.

Reply:
Remember, this is just to display the top 5 genes in the report.

Reply:
Yes but top 5 may be different depending on what you want to show.

Reply:
I like the percentage of cells. If you are against it, can you say why? In that case, we can change it to FC. 

I am introducing 2 parameters for the columns, so you may change the column name in the future.

Reply:
Now we would favor genes which are 100% expressed in the cluster but 0% expressed in the rest but which potentially with very slight changes (cluster 10 counts, rest 0 counts). I am not sure whether these are really marker genes.

In general, my main point here is the function. I would like to use it also for the DEGs. Therefore can you write it such that it can be used in both cases?

Reply:
Also, in the main script we write `We use the term "marker genes" to specifically describe genes that are up-regulated in cells of one cluster compared to the rest.`. Is that than still the case?

Reply:
I understand where you are going to. However, -first- I select on `p_adj_val_score`, so if there are more than the desired number, e.g. `n=5`, -then- I select based on the percentages. In this way, we pay attention to both big changes and the number of cells we see those changes in. 

With regard to your second question, yes I think so. Why not? We are looking at the table of marker genes, and sub-select from the table.

Reply:
I prefer to consider the percentages of genes as a second column to sort the genes. However, I am happy to sort on FC to display the top genes if you prefer that.

Reply:
Just change the function so that I can use it later please ....